src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -22,13 +22,297 @@
 #include <ripple/app/reporting/P2pProxy.h>
 #include <ripple/json/json_value.h>
 #include <ripple/net/RPCErr.h>
+#include <ripple/protocol/LedgerFormats.h>
+#include <ripple/protocol/SField.h>
+#include <ripple/protocol/TER.h>
+#include <ripple/protocol/TxFormats.h>
+#include <ripple/protocol/digest.h>
 #include <ripple/protocol/jss.h>
 #include <ripple/rpc/Context.h>
 #include <ripple/rpc/Role.h>
-#include <ripple/rpc/impl/TransactionSign.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <unordered_map>
 
 namespace ripple {
 
+namespace detail {
+
+class ServerDefinitions
+{
+private:
+    std::string
+    translate(std::string const& inp);
+
+    Json::Value
+    generate();
+
+    uint256 defsHash_;
+    Json::Value defs_;
+
+public:
+    ServerDefinitions() : defs_(generate()){};
```

Comment:
Why do we need a separate function `generate()` ? The constructor appears to be the only place, where this piece of code is executed.

Wouldn't the singleton pattern be more appropriate here? Do you intend to create one single instance of this class? I feel that would better communicate your intent

Reply:
Done in https://github.com/XRPLF/rippled/pull/4703/commits/205c0f5bc37b99d22becd640c4f587114f961edc